esmf_regrid/schemes.py

```diff
@@ -1270,6 +1425,7 @@ def __init__(
         use_src_mask=False,
         use_tgt_mask=False,
         tgt_location=None,
+        esmf_args=None,
```

Comment:
As discussed, I was unsure whether this needs to be a separate kwarg, since it effectively gets rolled into the generic "kwargs" parameter.
But I see now that keeping them separate makes sense, since we want `self.esmf_args` to record control details of the Esmpy usage.

However I do think it's now useful to add something to the docstring about usage of `esmf_args` and `kwargs`, to explain the context of how these are used in the operation of regridders, how+why the information is recorded for each regridder instance and stored/recovered in saving (io module).  
( Even if, as discussed, we're not at present quite sure of the last bit ! )